skimmer/skimmer_utils.py

```diff
@@ -96,10 +149,75 @@ def is_mc(events):
     return "Weight" in events.fields
 
 
-def apply_hem_veto(events):
-    """Apply HEM veto for the second part of 2018 data."""
+def apply_phi_spike_filter(events, year, jet_eta_branch_name="Jet_eta", jet_phi_branch_name="Jet_phi", reverse=False):
+    rad = 0.028816 # half the length of the diagonal of the eta-phi rectangular cell
+    rad *= 0.35 # the factor of 0.35 was optimized from the signal vs. background sensitivity study
 
-    def hem_veto(events):
+    eta_lead = None
+    eta_sub = None
+    phi_lead = None
+    phi_sub = None
+    if year == "2016":
+        eta_lead = [0.048,0.24,1.488,1.584,-1.008]
+        phi_lead = [-0.35,-0.35,-0.77,-0.77,-1.61]
+        eta_sub = [-1.2,-0.912,-0.912,-0.816,-0.72,-0.72,-0.528,-0.432,-0.336,-0.24,-0.24,-0.144,-0.144,-0.048,0.144,0.912,0.912,1.008,1.296,-1.584,-0.816,-0.72,-0.144,-0.048,-0.048,0.048,1.104,1.488]
+        phi_sub = [-1.19,2.03,3.01,-1.75,-2.17,-0.77,2.73,2.73,0.21,0.07,0.21,-2.59,0.77,0.91,1.75,1.75,2.87,0.63,-0.49,0.63,1.47,-2.31,0.07,-2.59,0.77,0.91,-3.15,2.73]
+    elif year == "2017":
+        eta_lead = [0.144,1.488,1.488,1.584,-0.624]
+        phi_lead = [-0.35,-0.77,-0.63,-0.77,0.91]
+        eta_sub = [-0.912,-0.912,-0.816,-0.72,-0.528,-0.336,-0.24,-0.24,-0.144,-0.144,-0.048,0.144,0.912,0.912,1.008,-1.2,-0.72,-0.72,-0.432,0.336,0.624,1.104,1.296]
+        phi_sub = [2.03,3.01,-1.75,-0.77,2.73,0.21,0.07,0.21,-2.59,0.77,0.91,1.75,1.75,2.87,0.63,-1.19,-2.31,-2.17,2.73,-0.77,-0.77,-3.15,-0.49]
+    elif year == "2018":
+        eta_lead = [1.488,1.488,1.584]
+        phi_lead = [-0.77,-0.63,-0.77]
+        eta_sub = [-1.584,-1.2,-0.912,-0.912,-0.816,-0.816,-0.72,-0.72,-0.528,-0.432,-0.336,-0.24,-0.24,-0.144,-0.144,-0.144,-0.048,-0.048,0.144,0.912,0.912,1.008,1.296,-0.72,1.104,1.488,1.776]
+        phi_sub = [0.63,-1.19,2.03,3.01,-1.75,-0.77,-2.17,-0.77,2.73,2.73,0.21,0.07,0.21,-2.59,0.07,0.77,0.77,0.91,1.75,1.75,2.87,0.63,-0.49,-2.31,-3.15,-0.21,0.77]
+    else:
+        raise ValueError("Invalid year")
+
+    eta_lead = nb.typed.List(eta_lead)
+    eta_sub = nb.typed.List(eta_sub)
+    phi_lead = nb.typed.List(phi_lead)
+    phi_sub = nb.typed.List(phi_sub)
+
+    jets_eta = getattr(events, jet_eta_branch_name)
+    jets_phi = getattr(events, jet_phi_branch_name)
+
+    builder = ak.ArrayBuilder()
+    phi_spike_filter = __get_phi_spike_filter(builder, eta_lead, phi_lead, eta_sub, phi_sub, rad, jets_eta, jets_phi, reverse=reverse).snapshot()
+
+    events = events[phi_spike_filter]
+
+    return events
+
+
+
+def apply_hem_filter(events, year, jet_pt_branch_name="Jet_pt", jet_eta_branch_name="Jet_eta",
```

Comment:
You added a new mandatory argument, which was not there before, this will break TreeMaker code. Can you make the `year` argument optional? Otherwise you can also put the year if the config itself:
```
    if year == "2018":
        events = skimmer_utils.apply_hem_veto(events)
        skimmer_utils.update_cut_flow(cut_flow, "HEMVeto", events)
```

Reply:
done